src/form.ts

```diff
@@ -0,0 +1,162 @@
+import { Terminal } from "@tuzig/xterm"
+import { FitAddon } from "xterm-addon-fit"
+
+export type Fields = Array<{
+    prompt:string,
+    default?:string,
+    values?:Array<string>,
+    validator?:(field: string) => string,
+}>
+
+export type Results = Array<string>
+
+export function openFormsTerminal(e: HTMLElement) {
+    const terminal = new Terminal({
+        cursorBlink: true,
+        cursorStyle: "block",
+        theme: window.terminal7.conf.theme,
+        fontFamily: "FiraCode",
+        fontSize: 14,
+        rendererType: "canvas",
+        convertEol: true,
+    })
+    terminal.open(e)
+    const fitAddon = new FitAddon()
+    terminal.loadAddon(fitAddon)
+    fitAddon.fit()
+    return terminal
+}
+
+export class Form {
+
+    field: string
+    i: number
+    e: HTMLElement
+    resolve: (value) => void
+    reject: (reason?) => void
+    fields: Fields
+    results: Results
+
+    constructor(fields: Fields) {
+        this.fields = fields
+    }
+
+    chooseFields(t: Terminal) {
+        const enabled = new Array(this.fields.length).fill(false)
+        let current = 0
+        return new Promise<Array<boolean>>((resolve, reject) => {
+            t.write("\n  Choose fields to edit:")
+            t.write("\n  (Press Enter to select/deselect, D when done, Escape to cancel)")
+            t.write("\n  " + this.fields.map(f => `[ ] ${f.prompt}: ${f.default}`).join('\n  '))
+            t.write("\x1B[4;4H") // move cursor to first field
+            const disposable = t.onKey(ev => {
+                const key = ev.domEvent.key
+                const char = !enabled[current] ? 'X' : ' '
+                switch (key) {
+                    case "Escape":
+                        reject()
+                        return
```

Comment:
Seems like you use both `break` and `return` in this event handler. They are both doing the same but it might confuse future developers. Unless there's a good reason to pay with the two, I suggest we use `break` in all the the cases